### PR TITLE
MedEx 6.0.0

### PR DESCRIPTION
**Pull request by ophthal**
Sign-up Bug Fix
Force full login to complete sign-up.

below added by bradymiller
Placed up for grabs demo for this PR here:
https://www.open-emr.org/wiki/index.php/Development_Demo#Gamma_-_Up_For_Grabs_Demo

### CODE REVIEW

File: interface/main/messages/save.php [hunk 1]
```diff
@@ -128,9 +128,9 @@
 								PHONE_country_code,LABELS_local,LABELS_choice)
 							VALUES (?,?,?,?,?,?,?,?,?,?)";
             sqlStatement($sqlINSERT, array($response['customer_id'], $response['API_key'], $_POST['new_email'], $facilities, $providers, "1", "1", "1", "1", "5160"));
-            sqlQuery("UPDATE `background_services` SET `active`='1',`execute_interval`='5', `require_once`='/library/MedEx/MedEx_background.php' WHERE `name`='MedEx'");
-
-            $info = $MedEx->login('1');
+            $query = "UPDATE `background_services` SET `active`='1',`execute_interval`='?, `require_once`='/library/MedEx/MedEx_background.php' WHERE `name`='MedEx'";
```
Comment (bradymiller) [inline interface/main/messages/save.php:131]: this query will likely fail since there is a lone apostrophe to the left of the `?` character

**Comment by ophthal** [comment]
Yep

File: interface/themes/tabs_style_full.scss [hunk 2]
```diff
@@ -120,7 +120,6 @@ body {
 }
 
 .tabControls {
-  background: var(--light);
```
Comment (stephenwaite) [inline interface/themes/tabs_style_full.scss:123]: hi @ophthal , the removal of this line is causing the tests to fail

**Comment by bradymiller** [comment]
Up For Grabs demo for this PR here: https://www.open-emr.org/wiki/index.php/Development_Demo#Gamma_-_Up_For_Grabs_Demo

File: interface/main/tabs/templates/tabs_template.php [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 <script type="text/html" id="tabs-controls">
     <div class="tabControls" data-bind="with: tabs">
-        <div class="tabNotchosen w-1">
```
Comment (bradymiller) [inline interface/main/tabs/templates/tabs_template.php:17]: hi @ophthal , removing this class is potentially problematic. Can this be left and then just address the styling of it?

File: interface/themes/tabs_style_full.scss (hunk 2 above)
Comment (bradymiller) [reply interface/themes/tabs_style_full.scss:123]: pretty sure e2e testing is failing because of the removal of the tabNotchosen class above

**Comment by ophthal** [comment]
Steve said the same thing but I am not sure why.
The issue comes down to styling not functionality.
tabControls here basically sets the background for every theme.
The class should still be in the themes right?

**Comment by stephenwaite** [comment]
when i remove `background: var(--light); ` in `tabs_style_full.scss` am unable to add a new user since it populates the Username with the existing admin:
Screenshot from 2021-03-08 07-30-07

File: interface/main/tabs/templates/tabs_template.php (hunk 3 above)
Comment (sjpadgett) [reply interface/main/tabs/templates/tabs_template.php:17]: This is because tabNotchosen is referenced by a visibility function.

Comment (stephenwaite) [reply interface/main/tabs/templates/tabs_template.php:17]: knockedout again 
knockout

**Comment by ophthal** [comment]
> when i remove `background: var(--light); ` in `tabs_style_full.scss` am unable to add a new user since it populates the 

Not sure how this blocks you from changing the text in the input box.  I just added a new user Fred using this branch in the demo server.
The next refresh of this server will have the last branch pushed - makes these tabs work well.  It evens deletes some !importants.  The first area (with the caret) still doesn't need a class but if it must tabsNoHover?

**Comment by ophthal** [comment]
What does that mean?

**Comment by stephenwaite** [comment]
sorry, Jerry had to help me with knockout the other day, all 2 lines of code had me baffled :)

**Comment by stephenwaite** [comment]
hi @ophthal, just pulled your branch and having the same user name issue, you can load an initial user but upon going back to add a second user it grabs your logged in user as the new username

**Comment by ophthal** [comment]
Why is that happening from a css issue in a tab directive - even a separate frame?
Also, what browser?
It is not happening here on a Mac/FF.

**Comment by ophthal** [comment]
I was looking at the manila theme trying to get it back to where it was.
I am used to the 3 dimensional view of this patient info/tabs section.  

For this comment though, this first section is not a tab.  Perhaps it should have the class “tabsNoHover” instead?
It works though without anything. 

I think the best way to see what these changes do it is to look at them and compare them to the dev tip without them.
Compare the full theme with Light to compact theme manila.


Full/Light




Compact/Manila


And it tests pretty well.


When you get to other schemes, ie other colors, there are !important overrides in the theme files to try and fix the color problems set in full/compact.
The fix is simple for those and it lies in the tabsNotChosen class.
Adding css opacity:75% and deleting transparency directives keeps the color styling.
Onmouseover then messes with tab display, but by deleting the lines in themes:

	 &:hover {
    		background: 0 0 !important; 
	}

It looks good.
I just pushed the branch with these changes. https://github.com/ophthal/openemr-1/tree/MedEx6Fix <https://github.com/ophthal/openemr-1/tree/MedEx6Fix>
Please test it out and see if it is acceptable…  


Ray

**Comment by stephenwaite** [comment]
chrome is choking on it for whatever reason

**Comment by ophthal** [comment]
I just tried on Chrome and Chromium (Mac though) and it works.  The field is pre-filled but changeable.
I cannot reproduce this problem.
So in Chrome for Windows the “Username” field is disabled?

**Comment by stephenwaite** [comment]
it's chome in gnu/linux in my case ubuntu 20 desktop, username get's prepopulated

**Comment by ophthal** [comment]
But is it changeable?

**Comment by stephenwaite** [comment]
yep

**Comment by ophthal** [comment]
The same thing is happening in demo 8 (https://eight.openemr.io/b/openemr/ <https://eight.openemr.io/b/openemr/>) so it is probably not this css issue but something else.

**Comment by sjpadgett** [comment]
Probably browser is doing autocomplete populating.

File: interface/main/tabs/templates/tabs_template.php (hunk 3 above)
Comment (sjpadgett) [reply interface/main/tabs/templates/tabs_template.php:17]: You still need to deal with this
            <div class="tabSpan bgcolor2" data-bind="click: tabClicked, css: {tabNotchosen: !visible()}">

**Comment by bradymiller** [comment]
hi @ophthal , let me know when you are done. Then I'll update it to get the automated tests working and get this into the codebase (holding the release of patch 1 until get this in :) ).

File: interface/main/tabs/templates/tabs_template.php (hunk 3 above)
Comment (sjpadgett) [reply interface/main/tabs/templates/tabs_template.php:17]: @bradymiller Prob won't break tabs but will error. Changing to tabsNoHover should be okay. It's only used in this one instance...

**Comment by bradymiller** [comment]
hi @ophthal , Just confirming this is ready to go (I think this is what you meant above, but unsure if another fix is pending). If so, then I will fix up the psr12 and testing stuff on this PR tomorrow and try to get the patch out. thanks! -brady

**Comment by ophthal** [comment]
Hi @bradymiller, I took another closer look at flow board and fixed more visibility/usability issues.  The Kiosk Mode now seems to work fine.  Fixes displaying/hiding various items as requested in Flow Board Globals and re-added Check-out time to Flow Board.  Happy to help and looking forward to upgrading to this release.

**Comment by bradymiller** [comment]
hi @ophthal , bringing this in. thanks for the fixes! Will hopefully get the patch out tomorrow.

**Comment by stephenwaite** [comment]
88220254-ba0afc00-cc30-11ea-9dd4-503ca1172e0e

**Pull request merged**